### PR TITLE
Refactor error response message handling

### PR DESCRIPTION
**Pull request by marcociarrocca**

### CODE REVIEW

**Comment by marcociarrocca** [comment]
> Looks good at a first look, I assume they pass?
Tests do pass, yes

**Comment by marcociarrocca** [comment]
@kerberjg please re-review latest commit

**Comment by marcociarrocca** [comment]
Added HttpServerService config validation

**Comment by kerberjg** [comment]
I think the last commit is outside the scope for this PR. Let's open a new one and continue the discussion there
As for the previous changes, I've reviewed them. Get back to me on those comments and let's merge this in soon

**Comment by marcociarrocca** [comment]
Reverted last commit as per your request

**Comment by kerberjg** [comment]
If you look above there's two comments pending your reply

**Comment by marcociarrocca** [comment]
I'm sorry, again, they reference nothing and I though I addressed them in [this commit](https://github.com/mekomidev/yagura-http/pull/3/commits/1f5a9af744e540a43b4c306a52c532d1569d8c3e), so, may I ask you to be a little more specific as to what I still need to fix?

File: test/httpApiLayer.test.ts [hunk 1]
```diff
@@ -238,7 +278,7 @@ describe('HttpApiLayer', () => {
                 return Promise.resolve(res);
             }
             public create(input: Partial<number>): Promise<CrudResponse<number>> {
-                const id = this.data.push(input) - 1;
+                // const id = this.data.push(input) - 1;
```
Comment (marcociarrocca) [inline test/httpApiLayer.test.ts:281]: test2

File: test/httpApiLayer.test.ts [hunk 2]
```diff
@@ -195,6 +210,30 @@ describe('HttpApiLayer', () => {
         expect(resDelete).to.have.status(200);
     });
 
+    it("should respond with the error\'s object type as the response\'s body", async () => {
```
[the remark this answers is not in the document]
Comment (marcociarrocca) [reply test/httpApiLayer.test.ts:222]: Added in fca1bfba922b38e5ff127dabec013f9a6b4b8db7

**Pull request merged**